frontend/src/components/manageUsers/ActiveUsers.tsx

```diff
@@ -0,0 +1,129 @@
+"use client";
+
+import type { UserDetail, UserRole } from "src/types/userTypes";
+
+import { useTranslations } from "next-intl";
+import React from "react";
+import { ErrorMessage, Table } from "@trussworks/react-uswds";
+
+export interface UsersTableProps {
+  roles: UserRole[];
+  tableDescription: string;
+  tableTitle: string;
+  users: UserDetail[];
+  busyUserId?: string;
+  isSubmitting?: boolean;
+  errorMessage?: string;
+  onRequestRoleChange: (userId: string, nextRoleId: string) => void;
+}
+
+function formatFullName(user: UserDetail): string {
+  if (!user || !user.first_name) return "-";
+
+  const parts = [user.first_name, user.middle_name, user.last_name].filter(
+    Boolean,
+  );
+
+  return parts.length > 0 ? parts.join(" ") : "-";
+}
+
+function getRoleId(role: UserRole): string {
+  return (role as { role_id: string }).role_id;
```

Comment:
??? `role_id` should already exist in UserRole, why is this necessary?

oh I think I see it - I think you can handle the typing within the parent function

```
  return match ? getRoleId(match as UserRole) : "";
```

?